microprofile/server/src/main/java/io/helidon/microprofile/server/Server.java

```diff
@@ -278,12 +279,40 @@ private SeContainer createContainter(ClassLoader classLoader) {
                                                               .asOptionalMap()
                                                               .orElse(CollectionsHelper.mapOf()));
             initializer.setProperties(props);
+
+            // add resource classes explicitly configured without CDI annotations
+            this.resourceClasses.stream()
+                    .filter(this::notACdiBean)
+                    .forEach(initializer::addBeanClasses);
```

Comment:
I think we need to make sure that if we have discovered class `X` but an instance of `X` is reported by `Application#getSingletons()` then probably we shouldn't add `X` here.

We also need to make sure we add the return value of `Application#getSingletons()`, if non-`null` and non-empty, to the CDI container.